src/mesh/region_set.cc

```diff
@@ -0,0 +1,315 @@
+				it_element != element_list.end();
+		        ++it_element) {
+			element_ids.push_back( *it_element );
+
+		}
+		mesh->modify_element_ids(region_id, element_ids);
+	}
+}
+
+
+
+const IT::Record & RegionSetFromElements::get_region_input_type()
+{
+    return IT::Record("From_Elements", "Region declared by name and enum of elements.")
+        .derive_from(RegionSetBase::get_input_type())
+		.declare_key("name", IT::String(), IT::Default::obligatory(),
+				"Label (name) of the region. Has to be unique in one mesh.\n")
+		.declare_key("id", IT::Integer(0), IT::Default::optional(),
+				"The ID of the region to which you assign label.")
+		.declare_key("element_list", IT::Array( IT::Integer(0) ), IT::Default::optional(),
+				"Specification of the region by the list of elements. This is not recomended")
+		.close();
+}
+
+
+
+const int RegionSetFromElements::registrar =
+		Input::register_class< RegionSetFromElements, const Input::Record &, Mesh * >("From_Elements") +
+		RegionSetFromElements::get_region_input_type().size();
+
+
+
+/*******************************************************************
+ * implementation of RegionSetBoundary
+ * Need new implementation, will be solved later.
+ */
+
+// RegionSetBoundary::RegionSetBoundary(const Input::Record &rec, Mesh *mesh) : RegionSetBase(mesh) {}
+
+// const IT::Record & RegionSetBoundary::get_region_input_type() {}
+
+//const int RegionSetBoundary::registrar =
+//		Input::register_class< RegionSetBoundary, const Input::Record &, Mesh * >("Region_Boundary") +
+//		RegionSetBoundary::get_region_input_type().size();
+
+
+
+/*******************************************************************
+ * implementation of RegionSetUnion
+ */
+
+RegionSetUnion::RegionSetUnion(const Input::Record &rec, Mesh *mesh)
+: RegionSetBase(mesh)
+{
+	RegionSet region_set;
+	string name_of_set = rec.val<string>("name");
+	Input::Iterator<Input::Array> region_ids = rec.find<Input::Array>("region_ids");
+	Input::Iterator<Input::Array> regions = rec.find<Input::Array>("regions");
+
+	if (region_ids) {
+		for (Input::Iterator<unsigned int> it_ids = region_ids->begin<unsigned int>();
+				it_ids != region_ids->end();
+		        ++it_ids) {
+			try {
+				Region reg = region_db_.find_id(*it_ids);
+				if (reg.is_valid()) {
+					if ( std::find(region_set.begin(), region_set.end(), reg)==region_set.end() ) {
+						region_set.push_back(reg); // add region if doesn't exist in set
+					}
+				} else {
+					xprintf(Warn, "Region with id %d doesn't exist. Skipping\n", (*it_ids));
+				}
+			} catch(RegionDB::ExcUniqueRegionId &e) {
+				e << region_ids->ei_address();
+				throw;
+			}
+		}
+	}
+
+	if (regions) {
+		std::vector<string> set_names = mesh->region_db().get_and_check_operands(*regions);
+		for (string set_name : set_names) {
+			region_set = region_db_.union_sets( region_set, set_name );
+		}
+	}
+
+	region_db_.add_set(name_of_set, region_set);
```

Comment:
Keep both keys "region_ids" and "regions" optional. Here check if the resulting region_set is not empty, possibly throw an input exception "Empty result of Union operation."